MdeModulePkg/Bus/Pci/PciBusDxe/PciEnumeratorSupport.c

```diff
@@ -2781,6 +2781,13 @@ IsPciDeviceRejected (
       //
       Mask      = 0xFFFFFFFC;
       TestValue = TestValue & Mask;
+      if ((TestValue != 0) && ((TestValue & 0xFFFF0000) == 0)) {
+        //
+        // IO Bar uses 16-bit decoding, pad for comparison
+        //
+        TestValue = TestValue | 0xFFFF0000;
+      }
+

```

Comment:
I don't think adding more magic checks is going to solve this: fundamentally, it is impossible to distinguish a 'rejected' BAR from one that has all writable bits set to `0x1` because that is where the decoded region lives.

And actually, poking bits in the BARs just to keep track of the software's decision that the device in question has been rejected seems a bit odd to begin with, and the fact that valid BARs are misidentified that were inherited from a preceding firmware stage by the light enumerator is clearly an oversight.

Please take a look at #11341 which proposes a more thorough approach